cluster.go

```diff
@@ -455,16 +502,58 @@ func (c *clusterClient) _pickMulti(multi []Completed) (retries *connretry, last
 	defer c.mu.RUnlock()
 
 	count := conncountp.Get(len(c.conns), len(c.conns))
-	for _, cmd := range multi {
-		if cmd.Slot() == cmds.InitSlot {
-			init = true
-			continue
+	var connIndexes []conn
+	if c.opt.SendToReplicas == nil {
+		for _, cmd := range multi {
+			if cmd.Slot() == cmds.InitSlot {
+				init = true
+				continue
+			}
+			p := c.pslots[cmd.Slot()]
+			if p == nil {
+				return nil, 0
+			}
+			count.m[p]++
 		}
-		p := c.slots[cmd.Slot()]
-		if p == nil {
-			return nil, 0
+	} else {
+		connIndexes = make([]conn, len(multi)) // lazy init
+		for i, cmd := range multi {
+			if cmd.Slot() == cmds.InitSlot {
+				init = true
+
+				var p conn
+				isSendToReplicas := c.opt.SendToReplicas(cmd)
+				if isSendToReplicas {
+					p = c.rslots[rand.Intn(int(cmds.InitSlot))]
+				} else {
+					p = c.pslots[rand.Intn(int(cmds.InitSlot))]
+				}
+
+				if p == nil {
+					return nil, 0
+				}
+
+				count.m[p]++
+				connIndexes[i] = p
```

Comment:
This part looks inconsistent with the previous implementation where we only do
```go
init = true
continue
```
because we would like to defer slot selection when we have commands with no slot in DoMulti().

Reply:
For example, when pipelining the `MULTI`, `SET k v`, `EXIPRE k 1000`, `EXEC` through DoMulti, we would choose the slot by `SET k v` and `EXIPRE k 1000`.

Reply:
there is test cases for this:

```
	t.Run("DoMulti Single Slot Read Operation And Write Operation + Init Slot", func(t *testing.T) {
		c1 := client.B().Get().Key("K1{a}").Build()
		c2 := client.B().Set().Key("K2{a}").Value("V2{a}").Build()
		c3 := client.B().Info().Build()
		resps := client.DoMulti(context.Background(), c1, c2, c3)
		if v, err := resps[0].ToString(); err != nil || v != "GET K1{a}" {
			t.Fatalf("unexpected response %v %v", v, err)
		}
		if v, err := resps[1].ToString(); err != nil || v != "SET K2{a} V2{a}" {
			t.Fatalf("unexpected response %v %v", v, err)
		}
		if v, err := resps[2].ToString(); err != nil || v != "INFO" {
			t.Fatalf("unexpected response %v %v", v, err)
		}
	})
```

Above case, "INFO" command is ignored. so i add connection to init slot command

Reply:
I don't get what you mean by providing the test case.

I mean, previously we intentionally did not select connections for `cmd.Slot() == cmds.InitSlot`. Therefore, I expect you would also write the same code of
```go
if cmd.Slot() == cmds.InitSlot {
	init = true
	continue
}
```
here.

Reply:
It is possible if multiple connections is selected and there is "InitSlot" command, then "InitSlot" command is ignored. 

For exmaple,
"GET K1{a}" command uses connection "A", "SET K2{a} V2{a}" command uses connection "B", and there is "InitSlot" command likes "INFO" together. original code ignore "INFO" because none of connections is assigned.

So if you want to changed to original code, above ignoring command is what you intended?

Reply:
I see the problem. But, we should only pick one connection if "InitSlot" commands are involved, otherwise commands like `MULTI` and `EXEC` will fail.

Reply:
OK. How about this? 
[eac17d5](https://github.com/redis/rueidis/pull/402/commits/eac17d5f2ab9afedfeab468c8529d1343365d72f)
When "InitSlot" command exist among multiple commands, use one of primary node only.

Reply:
> OK. How about this? [eac17d5](https://github.com/redis/rueidis/pull/402/commits/eac17d5f2ab9afedfeab468c8529d1343365d72f) When "InitSlot" command exist among multiple commands, use one of primary node only.

Hi @proost,

Sure. Using one of the primary nodes only sounds good to me.

However, the [eac17d5](https://github.com/redis/rueidis/pull/402/commits/eac17d5f2ab9afedfeab468c8529d1343365d72f) makes the _pickMulti becomes too complex, IMHO. How about we refactor it a little bit to first detect if there are "InitSlot" commands? If yes, we then just go to the original logic.

Reply:
[1178a7c](https://github.com/redis/rueidis/pull/402/commits/1178a7cfa1763dd2333525147020c5bf2ae9f708)

Sure! I changed it more simple